### ISSUE TITLE
ticklabel anchor cannot be determined

### ISSUE THREAD

**Issue by Peixiann** [matlab2tikz/matlab2tikz#955]
Hi there,

I have some problems when converting contourf, surf and quiver plots with matlab2tikz. When I implement the *.tikz file in Tex I get the following warning:

Package pgfplots Warning: the ticklabel anchor cannot be determined, the normal
vector -(0.0pt,-1.0pt) and the unit y vector (0.0pt,1.0pt) are almost parallel
(abs(cos(angle)) = 1.0pt)

An example of the compiled figure by tex looks as following:
![contourf_example](https://cloud.githubusercontent.com/assets/23662598/20534623/1d623e9e-b0e2-11e6-8dc9-096220a75ac6.jpg)

The Matlab code which generated this figure:
`    figure,
    contourf(X,Y,curlz,20,'LineStyle','none')
    hold on
    xlabel('x/h (-)');ylabel('y/h (-)')
    colorbar
    c = colorbar;
    c.Label.String = 'vorticity (-)';
    caxis([-25 5]);
`

I'm using Matlab R2016a, pgfplots 1.14, Matlab2tikz v1.1.0-0-g816f87 and my latex distribution is TexLive 2016. My operating system is windows 8.1

I don't get this error when creating line graphs. 
Hope someone can help. Thanks in advance.

**Comment by okomarov** [comment]
Can you please provide example inputs for `X`, `Y` and `curlz`?

**Comment by Peixiann** [comment]
Sure. They're all 127x127 double. So I only display a part of them:

`curlz(25:30,25:30)
ans =
       NaN       NaN       NaN       NaN       NaN       NaN
       NaN       NaN       NaN       NaN       NaN       NaN
   -2.9696   -2.3500   -3.2605   -1.7536   -3.6899   -3.4369
   -3.5453   -3.0136   -2.3730   -2.2996   -2.5990   -2.8765
   -3.0315   -2.4912   -2.3449   -2.4629   -2.6588   -2.6664
   -2.1164   -2.0990   -2.2437   -2.1892   -2.3460   -2.3178`

` X(25:30,25:30)
ans =
   -1.3025   -1.2873   -1.2720   -1.2568   -1.2415   -1.2263
   -1.3025   -1.2873   -1.2720   -1.2568   -1.2415   -1.2263
   -1.3025   -1.2873   -1.2720   -1.2568   -1.2415   -1.2263
   -1.3025   -1.2873   -1.2720   -1.2568   -1.2415   -1.2263
   -1.3025   -1.2873   -1.2720   -1.2568   -1.2415   -1.2263
   -1.3025   -1.2873   -1.2720   -1.2568   -1.2415   -1.2263`

`Y(25:30,25:30)
ans =
    0.0143    0.0143    0.0143    0.0143    0.0143    0.0143
    0.0295    0.0295    0.0295    0.0295    0.0295    0.0295
    0.0448    0.0448    0.0448    0.0448    0.0448    0.0448
    0.0600    0.0600    0.0600    0.0600    0.0600    0.0600
    0.0753    0.0753    0.0753    0.0753    0.0753    0.0753
    0.0905    0.0905    0.0905    0.0905    0.0905    0.0905`

**Comment by okomarov** [comment]
We need to be able reproduce your problem, so that won't do. In other words, I need to copy-paste your code and see where the problem is. You can re-create the data, simplify the problem or upload it somwwhere and share it then.

**Comment by Peixiann** [comment]
Okay, I created a simple script, which you should be able to copy and paste.
I just checked, I get the same error when compiling the *.tikz file in Tex

`close all; clc; clear;
X = 1:10;
Y = 1:10;
[X,Y]=meshgrid(X,Y);
curlz = rand(size(X)).*10;
figure, contourf(X,Y,curlz,20,'LineStyle','none'); hold on; xlabel('x/h (-)');ylabel('y/h (-)'); colorbar; c = colorbar; c.Label.String = 'vorticity (-)'; caxis([-25 5]);
matlab2tikz('contourplot_test.tikz', 'height', '\figureheight', 'width', '\figurewidth');
`

**Comment by okomarov** [comment]
I get the following:
![capture](https://cloud.githubusercontent.com/assets/3731173/20561834/745264ea-b178-11e6-9aea-592c137d5906.PNG)

Which is still a bit off, but we are aware of the problem of the base color. 

Can you try the `standalone` option:

`matlab2tikz('contourplot_test.tikz', 'standalone',true);`

**Comment by Peixiann** [comment]
When I try the `standalone` option I get a Latex error:

> ! LaTeX Error: Can be used only in preamble

for lines 7, 8, 9, 10, 11, 16 and 18. 
These lines in the .tikz file are:
\documentclass[tikz]{standalone}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{pgfplots}
\usepackage{grffile}
\usepackage{amsmath}
\begin{document}

I also get a Latex error for line 2126:

> ! You can't use `\end' in internal vertical mode.

Also the ticklabel warning is given.

Does the error have something to do with the colormap? I can also use a different colormap if this would solve the problem.

**Comment by egeerardyn** [comment]
The standalone option implies that you should run `pdflatex contourplot_test.tikz`  directly, without including it into your document. The output should be a PDF with only your figure.

That's a mechanism to make sure that the packages (not) loaded in your document do not interfere with TikZ/pgfplots.

**Comment by Peixiann** [comment]
Ah ok, now I understand what the standalone option is for.

This worked, I get the same type of figure as okomarov.  However, I'm still not able to include it in my document. Even when I remove all the packages in the preamble and copy-paste the preamble from the stand alone .tikz file I still get the warning:

`Package pgfplots Warning: the ticklabel anchor cannot be determined, the normal
vector -(0.0pt,-1.0pt) and the unit y vector (0.0pt,1.00002pt) are almost para
llel (abs(cos(angle)) = 1.00002pt)!`

**Comment by okomarov** [comment]
This is because you should not remove thsoe packages! Some are pgflopts specific, and you should include them.

**Comment by Peixiann** [comment]
I did not remove the preamble of the tikz file. I replaced the preamble in my document with the preamble of the standalone tikz file, and then tried to include the not-standalone tikz file in the document.  Which did not work.
I tried this because I assumed that the problem maybe had something to do with the (not) loaded packages.

However I still not understand why I get this warning. And why it also occurs with quiver and surf plots.

Anyway, is the funny looking contourf plot the best version matlab2tikz can produce? If it is I would have to find another solution, because this is not suitable to include in my document right now.

I would like to thank for the great and fast support. 
Still hoping to understand what is going on
